pkg/kubectl/cmd/describe.go

```diff
@@ -88,3 +95,39 @@ func RunDescribe(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []s
 	fmt.Fprintf(out, "%s\n", s)
 	return nil
 }
+
+func DescribeMatchingResources(mapper meta.RESTMapper, typer runtime.ObjectTyper, describer kubectl.Describer, f *cmdutil.Factory, namespace, rsrc, prefix string, out io.Writer) error {
+	r := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
+		NamespaceParam(namespace).DefaultNamespace().
+		ResourceTypeOrNameArgs(true, rsrc).
+		SingleResourceType().
+		Do()
+	obj, err := r.Object()
+	if err != nil {
+		return err
+	}
+	objs, err := f.List(obj)
```

Comment:
You shouldn't need this, the object can be genetically treated as a list by the resource builder.  Add Flatten to the builder and invoke Infos() or the Visit on the result.  Infos has name already (or should).